#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.0008657269065851993 -0.9375356063383918 -0.3478882541343231 3.154432741430381
  position -0.00610151730975997 2.9327801226024506 -3.3503314904563313
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2 2
}
Robot {
  translation 0.11966651448770305 0.08048781130466207 0.18172156603611397
  rotation 0.0008991795714439398 0.0009069320299307954 0.9999991844748631 -0.005861308207615254
  scale 1.0 1.0 1.0
  children [
    Compass {
    }
    GPS {
    }
    DEF base_link Shape {
      appearance PBRAppearance {
        baseColor 0.0659342 0.501793 0.00683604
        metalness 0
      }
      geometry Box {
        size 0.38 0.03 0.15
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -3.1382757541227665e-07
        axis 0 0 -1
        anchor 0.1 0 0.08
      }
      device [
        RotationalMotor {
          name "right_front_wheel"
        }
      ]
      endPoint Solid {
        translation 0.1 0 0.08
        rotation 0.9999999999999755 1.5691378880563836e-07 1.5703879174589135e-07 1.5700000000000205
        children [
          DEF wheel_shape Shape {
            appearance PBRAppearance {
              baseColor 0.8 0.5 0
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.055
            }
          }
        ]
        name "right_front_wheel"
        boundingObject USE wheel_shape
        physics DEF wheel_physics Physics {
        }
        locked TRUE
        linearVelocity 7.81862113031765e-09 3.965559322651446e-07 2.653146469092142e-12
        angularVelocity -1.363037394980737e-10 5.600026852007554e-12 -1.7337211740023945e-07
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 6.186800324818075e-08
        axis 0 0 -1
        anchor -0.02 0 0.08
      }
      device [
        RotationalMotor {
          name "right_mid_wheel"
        }
      ]
      endPoint Solid {
        translation -0.02 0 0.08
        rotation 0.999999999999999 -3.093400162402324e-08 -3.095864501192145e-08 1.569999999999998
        children [
          USE wheel_shape
        ]
        name "right_mid_wheel"
        boundingObject USE wheel_shape
        physics USE wheel_physics
        locked TRUE
        linearVelocity 1.4399043443622759e-08 2.9986686535053686e-07 -1.0462432606699643e-11
        angularVelocity -1.2666227356615226e-10 6.016136633392042e-12 -1.7444334465238877e-07
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 2.5216249142868394e-07
        axis 0 0 -1
        anchor -0.14 0 0.08
      }
      device [
        RotationalMotor {
          name "right_rear_wheel"
        }
      ]
      endPoint Solid {
        translation -0.14 0 0.08
        rotation 0.9999999999999842 -1.2608124587141243e-07 -1.2618168766475822e-07 1.5700000000000107
        children [
          USE wheel_shape
        ]
        name "right_rear_wheel"
        boundingObject USE wheel_shape
        physics USE wheel_physics
        locked TRUE
        linearVelocity 2.0979465760166557e-08 2.031777995879612e-07 -2.3577917216024845e-11
        angularVelocity -1.1702203962636762e-10 6.4322160511824955e-12 -1.7551457174836255e-07
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -3.139535546339401e-07
        axis 0 0 -1
        anchor 0.1 0 -0.08
      }
      device [
        RotationalMotor {
          name "left_front_wheel"
        }
      ]
      endPoint Solid {
        translation 0.09999999999999999 -2.5162922253510527e-17 -0.08
        rotation 0.9999999999999754 1.5697677739550338e-07 1.5710183546381627e-07 1.570000000000016
        children [
          USE wheel_shape
        ]
        name "left_front_wheel"
        boundingObject USE wheel_shape
        physics USE wheel_physics
        locked TRUE
        linearVelocity 7.790491326275998e-09 3.966818797115627e-07 2.6870068705403907e-12
        angularVelocity -1.3631661424708647e-10 5.598274330156486e-12 -1.7336753843028817e-07
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 6.174202399371703e-08
        axis 0 0 -1
        anchor -0.02 0 -0.08
      }
      device [
        RotationalMotor {
          name "left_mid_wheel"
        }
      ]
      endPoint Solid {
        translation -0.02 0 -0.08
        rotation 0.9999999999999991 -3.087101199701189e-08 -3.089560520394757e-08 1.569999999999999
        children [
          USE wheel_shape
        ]
        name "left_mid_wheel"
        boundingObject USE wheel_shape
        physics USE wheel_physics
        locked TRUE
        linearVelocity 1.437091367800307e-08 2.999928113791824e-07 -1.0428333974340034e-11
        angularVelocity -1.2667434346759364e-10 6.014365390081956e-12 -1.7443876582137278e-07
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 2.519672898156875e-07
        axis 0 0 -1
        anchor -0.14 0 -0.08
      }
      device [
        RotationalMotor {
          name "left_rear_wheel"
        }
      ]
      endPoint Solid {
        translation -0.14 0 -0.08
        rotation 0.9999999999999842 -1.259836454575937e-07 -1.260840083198006e-07 1.570000000000014
        children [
          USE wheel_shape
        ]
        name "left_rear_wheel"
        boundingObject USE wheel_shape
        physics USE wheel_physics
        locked TRUE
        linearVelocity 2.095133593476068e-08 2.033037475661177e-07 -2.354406270343653e-11
        angularVelocity -1.170397434745435e-10 6.430435868462867e-12 -1.755099926818211e-07
      }
    }
    DistanceSensor {
      translation 0.19 -0.0075 0.075
      children [
        DEF ir_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "right_outer_sensor"
      boundingObject USE ir_shape
      physics DEF ir_physics Physics {
      }
      locked TRUE
    }
    DistanceSensor {
      translation 0.19 -0.0075 0.025
      children [
        DEF ir_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "right_mid_sensor"
      boundingObject USE ir_shape
      physics DEF ir_physics Physics {
      }
      locked TRUE
    }
    DistanceSensor {
      translation 0.19 -0.0075 -0.075
      children [
        DEF ir_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "left_outer_sensor"
      boundingObject USE ir_shape
      physics DEF ir_physics Physics {
      }
      locked TRUE
    }
    DistanceSensor {
      translation 0.19 -0.0075 -0.025
      children [
        DEF ir_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "left_mid_sensor"
      boundingObject USE ir_shape
      physics DEF ir_physics Physics {
      }
      locked TRUE
    }
    DistanceSensor {
      translation 0.17 -0.0075 0.0749601
      rotation 0 1 0 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "right_side_sensor"
      boundingObject USE ir_shape
      physics USE ir_physics
      locked TRUE
    }
    DistanceSensor {
      translation 0.17 -0.0075 -0.0749601
      rotation 0 1 0 1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "left_side_sensor"
      boundingObject USE ir_shape
      physics USE ir_physics
      locked TRUE
    }
    Camera {
      translation 0.13 0.18 -1.16711e-09
      rotation 0 1 0 -1.57
      children [
        DEF camera_shape Shape {
          appearance PBRAppearance {
            baseColor 1 0.999969 0.999985
            metalness 0
          }
          geometry Cylinder {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "front_camera"
      boundingObject USE camera_shape
      physics Physics {
      }
      locked TRUE
    }
    Solid {
      translation 0 0.05 -0.065
      children [
        DEF basket_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.37 0.1 0.01
          }
        }
      ]
      name "left basket"
      boundingObject USE basket_shape
      physics Physics {
      }
      locked TRUE
    }
    Solid {
      translation 0 0.05 0.065
      children [
        DEF basket_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.37 0.1 0.01
          }
        }
      ]
      name "right basket"
      boundingObject USE basket_shape
      physics Physics {
      }
      locked TRUE
    }
    Solid {
      translation -0.18 0.05 0
      rotation 0 1 0 1.57
      children [
        DEF basket_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.14 0.1 0.01
          }
        }
      ]
      name "back basket"
      boundingObject USE basket_shape
      physics Physics {
      }
      locked TRUE
    }
    Solid {
      translation 0.18 0.05 0
      rotation 0 1 0 1.57
      children [
        DEF basket_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.14 0.1 0.01
          }
        }
      ]
      name "front basket"
      boundingObject USE basket_shape
      physics Physics {
      }
      locked TRUE
    }
    Solid {
      translation 0.15 0.05 0
      rotation 0 1 0 1.57
      children [
        DEF basket_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.14 0.1 0.05
          }
        }
      ]
      name "motor_box"
      boundingObject USE basket_shape
      physics Physics {
      }
      locked TRUE
    }
    Solid {
      translation 0.125 0.1 0
      rotation 0 1 0 1.57
      children [
        DEF basket_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.02 0.2 0.02
          }
        }
      ]
      name "camera_bar"
      boundingObject USE basket_shape
      physics Physics {
      }
      locked TRUE
    }
  ]
  name "GardenRobot"
  boundingObject USE base_link
  physics Physics {
  }
  controller "gps_controller"
  linearVelocity 2.1294776838792062e-08 9.349248718804983e-08 -1.5578580984953162e-11
  angularVelocity -1.174546672780249e-10 5.9136164876866395e-12 -1.744421399992496e-07
}
